dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
     riders.rating as rider_rating
-
-from trips
-left join users as drivers
+  from trips
+  left join users as drivers
     on trips.driver_id = drivers.user_id
-left join users as riders
+  left join users as riders
     on trips.rider_id = riders.user_id
 
-```
+  ```
+
+- ### Example SQL
+  ```sql
+  with
+
+  my_data as (
+      select * from {{ ref('my_data') }}
+      where not is_deleted
+  ),
+
+  some_cte as (
+      select * from {{ ref('some_cte') }}
+  ),
+
+  some_cte_agg as (
+      select
+          id,
+          sum(field_4) as total_field_4,
+          max(field_5) as max_field_5
+      from some_cte
+      group by 1
+  ),
+
+  final as (
+      select [distinct]
+          my_data.field_1,
+          my_data.field_2,
+          my_data.field_3,
+
+          -- use line breaks to visually separate calculations into blocks
+          case
+              when my_data.cancellation_date is null
+                and my_data.expiration_date is not null
+                then expiration_data
+              when my_data.cancellation_date is null
+                then my_data.start_date + 7
+              else my_data.cancellation_date
+          end as cancellation_date,
+
+          some_cte_agg.total_field_4,
+          some_cte_agg.max_field_5
+      from my_data
+      left join some_cte_agg  
+          on my_data.id = some_cte_agg.id
+      where 
+        my_data.field_1 = 'abc'
+        and (
+            my_data.field_2 = 'def'
+            or my_data.field_2 = 'ghi'
+        )
+      having count(*) > 1
+  )
+
+  select * from final
+  ```
 
 ## YAML style guide
 
-* Indents should be two spaces
-* List items should be indented
-* Use a new line to separate list items that are dictionaries where appropriate
-* Lines of YAML should be no longer than 80 characters.
-
-### Example YAML
-```yaml
-version: 2
-
-models:
-  - name: events
-    columns:
-      - name: event_id
-        description: This is a unique identifier for the event
-        tests:
-          - unique
-          - not_null
-
-      - name: event_time
-        description: "When the event occurred in UTC (eg. 2018-01-01 12:00:00)"
-        tests:
-          - not_null
-
-      - name: user_id
-        description: The ID of the user who recorded the event
-        tests:
-          - not_null
-          - relationships:
-              to: ref('users')
-              field: id
-```
+- Every subdirectory contains their own `.yml` file(s) which contain configurations for the models within the subdirectory.
+
+- A separate `.yml` file is created per top-level configuration (`sources`, `models`) that applies to the models within the subdirectory. 
+
+- YAML files should be prefixed with an underscore to keep it at the top of the subdirectory.  
+  Example: `__sources_core.yml`, `_models_core.yml`
+
+- YAML files should be named for the top-level configuration it contains.  
+  Examples: `_sources_core.yml`, `_models_core.yml`
+
+- Indents should use four spaces.

```

Comment:
the example below uses two spaces instead of 4. I think we should update this to be 2 spaces

Reply:
+1 thought we prefer 2 for yml and 4 for SQL (and by extension Jinja)

Reply:
I think we need another vote on this because I'm terrible at finding conversations. There was a convo / vote on this, and I thought it was 2 all around. But yes, I think yml is definitely 2 - not sure what the vote is on SQL anymore! Guessing that because I changed the code within the examples, that there was a vote for 2 on SQL as well.

Reply:
My vote is however many spaces tab is for both... lazy I guess 🤷🏻‍♂️

Reply:
I don't really care for SQL (2 or 4), but definitely 2 for YML

Reply:
Fixed [here](https://github.com/dbt-labs/corp/pull/63/commits/b24fa76bae298c05105001532ad70c4b885a0192)!